apps/tangle-dapp/data/liquidStaking/helper.ts

```diff
@@ -0,0 +1,322 @@
+    if (!identity[1].unwrap()[0]) {
+      return;
+    }
+
+    const validator = identity[0].args[0];
+    const info = identity[1].unwrap()[0].info;
+    let displayName = validator.toString();
+    if (info) {
+      const displayData = info['display'];
+      if (displayData.isNone) return null;
+      const displayDataObject: { raw?: string } = JSON.parse(
+        displayData.toString(),
+      );
+      if (displayDataObject.raw !== undefined) {
+        const hexString = displayDataObject.raw;
+        displayName = Buffer.from(hexString.slice(2), 'hex').toString('utf8');
+      }
+    }
+    map.set(validator.toString(), displayName);
+  });
+  return map;
+};
+
+export const fetchMappedValidatorsTotalValueStaked = async (
+  rpcEndpoint: string,
+): Promise<Map<string, BN>> => {
+  const api = await getApiPromise(rpcEndpoint);
+
+  if (!api.query.staking) {
+    throw new Error('staking pallet is not available in the runtime');
+  }
+
+  const activeEra = await api.query.staking.activeEra();
+  const activeEraIndex = activeEra.unwrap().index;
+  const validators = await fetchValidators(rpcEndpoint);
+  const stakerPromises = validators.map(async (val) => {
+    const erasStakers = await api.query.staking.erasStakersOverview(
+      activeEraIndex,
+      val,
+    );
+    const exposure = erasStakers.unwrap();
+    const totalStaked = exposure.total.unwrap().toBn() || BN_ZERO;
+    return {
+      validator: val.toString(),
+      totalStaked,
+    };
+  });
+  const stakers = await Promise.all(stakerPromises);
+  const map = new Map<string, BN>();
+  stakers.forEach((staker) => {
+    map.set(staker.validator, staker.totalStaked);
+  });
+  return map;
+};
+
+export const fetchMappedValidatorsCommission = async (
+  rpcEndpoint: string,
+): Promise<Map<string, BN>> => {
+  const api = await getApiPromise(rpcEndpoint);
+  const validators = await fetchValidators(rpcEndpoint);
+
+  if (!api.query.staking) {
+    throw new Error('staking pallet is not available in the runtime');
+  }
+
+  const commissionPromises = validators.map(async (val) => {
+    const prefs = await api.query.staking.validators(val);
+    return {
+      validator: val.toString(),
+      commission: prefs.commission.unwrap(),
+    };
+  });
+  const commissions = await Promise.all(commissionPromises);
+  const map = new Map<string, BN>();
+  commissions.forEach((commission) => {
+    map.set(commission.validator, commission.commission);
+  });
+  return map;
+};
+
+export const fetchChainDecimals = async (
+  rpcEndpoint: string,
+): Promise<number> => {
+  const api = await getApiPromise(rpcEndpoint);
+  const chainDecimals = await api.registry.chainDecimals;
+  return chainDecimals.length > 0 ? chainDecimals[0] : 18;
+};
+
+export const fetchTokenSymbol = async (
+  rpcEndpoint: string,
+): Promise<string> => {
+  const api = await getApiPromise(rpcEndpoint);
+  const tokenSymbol = api.registry.chainTokens[0];
+  return tokenSymbol;
+};
```

Comment:
If we know the chains where we're calling this, might be better to simply hard code the chain decimals and token symbols, since they're extremely unlikely to change anytime soon, so the burden of having to request this often might outweight simply hard coding it